### PR TITLE
Remove redundant/unused logic in useActiveAccount

### PR DESCRIPTION
**Pull request by swain**
## Changes
This PR simplifies a number of pieces of logic in `useActiveAccount` that seemed to be no longer useful.

I've highlighted each change below via a comment -- review appreciated.

### CODE REVIEW

File: src/hooks/useActiveAccount.tsx [hunk 1]
```diff
@@ -23,15 +22,13 @@ export type ActiveAccountProps = {
 };
 
 export type ActiveAccountContextProps = ActiveAccountProps & {
-  refetch: () => Promise<QueryObserverResult<Account[], unknown>>;
```
Comment (swain) [inline src/hooks/useActiveAccount.tsx:26]: `refetch` was not being used outside of the provider

File: src/hooks/useActiveAccount.tsx [hunk 2]
```diff
@@ -23,15 +22,13 @@ export type ActiveAccountProps = {
...
-  setActiveAccountId: (accountId: string) => Promise<void>;
+  setActiveAccountId: (accountId: string) => void;
```
Comment (swain) [inline src/hooks/useActiveAccount.tsx:25]: There was no need for this function to be `async` -- it's just a wrapper around a `setState` call

File: src/hooks/useActiveAccount.tsx [hunk 3]
```diff
@@ -120,43 +116,28 @@ export const ActiveAccountContextProvider = ({
...
   }, [activeAccount?.account?.id, setStoredAccountId]);
 
-  // Clear selected account when
-  // we've detected that the userId has changed
-  useEffect(() => {
-    if (userId !== previousUserId) {
-      accountsResult.refetch();
-      setPreviousUserId(userId);
-    }
-  }, [previousUserId, userId, accountsResult]);
```
Comment (swain) [inline src/hooks/useActiveAccount.tsx:130]: This `useEffect` was actually not necessary (which made "previous user id" tracking unnecessary as well). Why: the entire react-query `QueryClient` gets cleared wiped when the user changes. When that happens, a refetch of the query is triggered automatically.

File: src/hooks/useActiveAccount.tsx [hunk 4]
```diff
@@ -120,43 +109,28 @@ export const ActiveAccountContextProvider = ({
...
-  }, [accountsResult]);
-
   // Handle invite accept
   useEffect(() => {
     const listener = async (acceptedInvite: ProjectInvite) => {
-      cache.invalidateQueries({ 'GET /v1/accounts': 'all' });
-      await refetch();
-      await setActiveAccountId(acceptedInvite.account);
+      cache.resetQueries({ 'GET /v1/accounts': 'all' });
+      setSelectedId(acceptedInvite.account);
```
Comment (swain) [inline src/hooks/useActiveAccount.tsx:120]: `resetQueries` is a "hard reset" -- it wipes the current value from the cache and puts us back into the "initial loading" state. By doing this, we avoid needing to refetch, and more clearly signal "we're starting over"

File: src/hooks/useActiveAccount.tsx [hunk 5]
```diff
@@ -19,19 +18,16 @@ export type ActiveAccountProps = {
   account?: Account;
   accountsWithProduct?: Account[];
   accountHeaders?: Record<string, string>;
-  trialExpired?: boolean;
```
Comment (swain) [inline src/hooks/useActiveAccount.tsx:22]: This `trialExpired` property was not referenced anywhere. Seems like it's fallen out of use.

File: src/hooks/useActiveAccount.tsx (hunk 3 above)
Comment (bruce-glazier) [reply src/hooks/useActiveAccount.tsx:130]: > the entire react-query QueryClient gets cleared wiped when the user changes

That wasn't always the case. Are you certain it is setup that way?

**Review by bruce-glazier — APPROVED** [review]

**Pull request merged**